packages/models/src/tenant/tenant.ts

```diff
@@ -25,10 +25,20 @@ export const TenantFeatureCertifier = z.object({
   type: z.literal("PersistentCertifier"),
   certifierId: z.string(),
 });
-
 export type TenantFeatureCertifier = z.infer<typeof TenantFeatureCertifier>;
 
-export const TenantFeature = TenantFeatureCertifier; // It will be extended with other features, we will use this union to discriminate them
+export const TenantFeatureDelegatedProducer = z.object({
+  type: z.literal("DelegatedProducer"),
+  availabilityTimestamp: z.coerce.date(),
+});
+export type TenantFeatureDelegatedProducer = z.infer<
+  typeof TenantFeatureDelegatedProducer
+>;
+
+export const TenantFeature = z.discriminatedUnion("type", [
+  TenantFeatureCertifier,
+  TenantFeatureDelegatedProducer,
+]);
```

Comment:
now is more clear, anyway I suggest to use different naming convention instead of features for something related to a person Tenant.